llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp

```diff
@@ -650,7 +643,7 @@ struct RISCVOperand final : public MCParsedAsmOperand {
     RISCVMCExpr::Specifier VK = RISCVMCExpr::VK_None;
     if (!isImm())
       return false;
-    bool IsConstantImm = evaluateConstantImm(getImm(), Imm, VK);
+    bool IsConstantImm = evaluateConstantImm(getImm(), Imm);
     if (VK == RISCVMCExpr::VK_LO || VK == RISCVMCExpr::VK_PCREL_LO ||
```

Comment:
Nit: This `if` is always `false`, and I think `VK` is always none by the time you get to (new) line 654.